CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.cc

```diff
@@ -0,0 +1,432 @@
+    toCompressStream << "  WEIGHTAMP  " << dec << iGr << std::endl;
+    for (long unsigned int i = 0; i < ampWeights[iGr].size(); i++) {
+      if (ampWeights[iGr][i] < 0)
+        toCompressStream << "-0x" << std::hex << abs(ampWeights[iGr][i]) << " ";
+      else
+        toCompressStream << "0x" << std::hex << ampWeights[iGr][i] << " ";
+    }
+    toCompressStream << "\n";
+  }
+  toCompressStream << "\n";
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+
+  for (unsigned int iGr = 0; iGr < nWeightGroups_; iGr++) {
+    toCompressStream << "WEIGHTTIME " << dec << iGr << std::endl;
+    for (long unsigned int i = 0; i < timeWeights[iGr].size(); i++) {
+      if (timeWeights[iGr][i] < 0)
+        toCompressStream << "-0x" << std::hex << abs(timeWeights[iGr][i]) << " ";
+      else
+        toCompressStream << "0x" << std::hex << timeWeights[iGr][i] << " ";
+    }
+    toCompressStream << "\n";
+  }
+
+  toCompressStream << "\n";
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+
+  //  fill  map between xTals and groups. If each xTal is a group there is a one-to-one map
+  const std::vector<DetId>& ebCells = theBarrelGeometry->getValidDetIds(DetId::Ecal, EcalBarrel);
+  std::map<int, int> mapXtalToGroup;
+
+  int iGroup = 0;
+  for (const auto& it : ebCells) {
+    EBDetId id(it);
+    std::pair<int, int> xTalToGroup(id.rawId(), iGroup);
+    mapXtalToGroup.insert(xTalToGroup);
+    iGroup++;
+  }
+
+  //write to file
+
+  for (std::map<int, int>::const_iterator it = mapXtalToGroup.begin(); it != mapXtalToGroup.end(); it++) {
+    toCompressStream << "CRYSTAL " << dec << it->first << std::endl;
+    toCompressStream << it->second << std::endl;
+  }
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+
+  /////////////////////////////////////
+
+  for (const auto& it : ebCells) {
+    EBDetId id(it);
+    toCompressStream << "LINCONST " << dec << id.rawId() << std::endl;
+    double theta = theBarrelGeometry->getGeometry(id)->getPosition().theta();
+    EcalLiteDTUPedestalsMap::const_iterator itped = theEcalTPPedestals->getMap().find(id);
+
+    if (itped != theEcalTPPedestals->end()) {
+      peds = &(*itped);
+
+    } else {
+      edm::LogError("EcalEBPhase2TPParamProducer") << " could not find EcalLiteDTUPedestal entry for " << id;
+      throw cms::Exception("could not find pedestals");
+    }
+
+    int shift, mult;
+    double calibCoeff = 1.;
+    bool ok;
+    for (unsigned int i = 0; i < ecalPh2::NGAINS; ++i) {
+      ok = computeLinearizerParam(theta, ecalph2::gains[1 - i], calibCoeff, shift, mult);
```

Comment:
This can only work because `ecalPh2::NGAINS` is 2.
If you want to remain independent on the exact number (i.e. do not hardcode it) you should
```suggestion
      ok = computeLinearizerParam(theta, ecalph2::gains[ecalPh2::NGAINS -1 - i], calibCoeff, shift, mult);
```